### PR TITLE
enh(UI): Rework detail panel chip to add action on hover Chip

### PR DESCRIPTION
**Pull request by Jordan91**
## Description

Rework detail panel chip to add action on hover Chip

**Fixes** # 

https://www.loom.com/share/0b3bd76d1cf04cfcb00da09b00f17f88

## Type of change

- [x] New functionality (non-breaking change)

## Target serie

- [ ] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

### CODE REVIEW

**Comment by adr-mo** [comment]
New commit added for the backend part

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            <Tooltip title={group.name}>
+              <Typography
+                className={classes.groupChipAction}
+                style={{ color: isHovered ? 'transparent' : 'unset' }}
+                variant="body2"
+              >
+                {group.name}
+              </Typography>
+            </Tooltip>
+            {isHovered === true && (
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx:83]:
```suggestion
            {isHovered && (
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                  }}
+                >
+                  <SettingsIcon fontSize="small" />
+                </IconButton>
+              </Grid>
+            )}
+          </div>
+        }
+        onMouseEnter={(): void => setIsHovered(true)}
+        onMouseLeave={(): void => setIsHovered(false)}
```
[the remark this answers is not in the document]
Comment (Jordan91) [reply www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx:106]: Function like getChipStyle ?

[the remark this answers is not in the document]
Comment (Jordan91) [reply www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx:106]: or getChipDisplay

**Review by kduret — DISMISSED** [review]

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/DetailsLine.tsx [hunk 3]
```diff
@@ -8,7 +8,6 @@ import makeStyles from '@mui/styles/makeStyles';
 const useStyles = makeStyles((theme) => ({
   lineText: {
     fontSize: theme.typography.body2.fontSize,
-    overflow: 'hidden',
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/DetailsLine.tsx:11]: Did you check that this change doesn't impact negatively whe the line is bigger than the tile?

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/Groups.tsx [hunk 4]
```diff
@@ -1,46 +1,70 @@
+/* eslint-disable react/jsx-key */
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/Groups.tsx:1]: This shouldn't be disabled

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/Groups.tsx [hunk 5]
```diff
@@ -1,46 +1,70 @@
...
+import { ResourceType } from '../../../../models';
+
+import GroupChip from './GroupChip';
 
 interface Props {
-  details: ResourceDetails;
+  details: ResourceDetails | undefined;
 }
 
+const useStyles = makeStyles((theme) => ({
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/Groups.tsx:21]: Most of the classes aren't actually used

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx [hunk 6]
```diff
@@ -21,25 +21,26 @@ const useStyles = makeStyles((theme) => ({
...
+    display: 'flex',
+    gap: theme.spacing(0.25),
+    gridArea: '1/1',
   },
   groupChipLabel: {
     display: 'grid',
     justifyItems: 'center',
     minWidth: theme.spacing(7),
     overflow: 'hidden',
+    textOverflow: 'ellipsis',
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx:38]: the 'group' prefices seem redundant to me

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx [hunk 7]
```diff
@@ -21,25 +21,26 @@ const useStyles = makeStyles((theme) => ({
...
   },
   groups: {
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx:40]: This isn't used

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/Groups.tsx [hunk 8]
```diff
@@ -1,46 +1,48 @@
+/* eslint-disable react/jsx-key */
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/Groups.tsx:1]:
```suggestion
```

Comment (Jordan91) [reply www/front_src/src/Resources/Details/tabs/Details/DetailsCard/Groups.tsx:1]: i deleted this before pushing , command  eslint:fix can re-add this rules ?

File: src/Centreon/Domain/Contact/Contact.php [hunk 9]
```diff
@@ -61,6 +61,10 @@ class Contact implements UserInterface, ContactInterface
     public const ROLE_REPORTING_DASHBOARD_SERVICES = 'ROLE_REPORTING_DASHBOARD_SERVICES_RW';
     public const ROLE_CONFIGURATION_MONITORING_SERVER_READ_WRITE = 'ROLE_CONFIGURATION_POLLERS_POLLERS_RW';
     public const ROLE_CONFIGURATION_MONITORING_SERVER_READ = 'ROLE_CONFIGURATION_POLLERS_POLLERS_R';
+    public const ROLE_CONFIGURATION_HOSTS_HOST_GROUPS_READ_WRITE = 'ROLE_CONFIGURATION_HOSTS_HOST_GROUPS_RW';
+    public const ROLE_CONFIGURATION_HOSTS_HOST_GROUPS_READ = 'ROLE_CONFIGURATION_HOSTS_HOST_GROUPS_R';
+    public const ROLE_CONFIGURATION_SERVICES_SERVICE_GROUPS_READ_WRITE = 'ROLE_CONFIGURATION_SERVICES_SERVICE_GROUPS_RW';
```
Comment (kduret) [inline src/Centreon/Domain/Contact/Contact.php:66]:
```suggestion
    public const ROLE_CONFIGURATION_SERVICES_SERVICE_GROUPS_READ_WRITE =
        'ROLE_CONFIGURATION_SERVICES_SERVICE_GROUPS_RW';
```

**Review by kduret — DISMISSED** [review]

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+    setCriteriaAndNewFilter({
+      name: type,
+      value: [group],
+    });
+  };
+  const onMouseEnter = (): void => {
+    setIsHovered(true);
+  };
+
+  const onMouseLeave = (): void => {
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx:67]:
```suggestion
  const mouseEnter = (): void => {
    setIsHovered(true);
  };

  const mouseLeave = (): void => {
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+    <Grid item className={classes.chip} key={group.id}>
+      <Chip
+        aria-label={`${group.name} Chip`}
+        color="primary"
+        label={
+          <div className={classes.chipLabel}>
+            <Tooltip title={group.name}>
+              <Typography
+                className={classes.chipAction}
+                style={{ color: isHovered ? 'transparent' : 'unset' }}
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx:81]: I think, you can set the color via className

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+              >
+                {group.name}
+              </Typography>
+            </Tooltip>
+            {isHovered && (
+              <Grid className={classes.chipHover}>
+                <IconButton
+                  aria-label={`${group.name} Filter`}
+                  size="small"
+                  style={{ color: theme.palette.common.white }}
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx:92]: You can set the color via className

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+                  title={t(labelFilter)}
+                  onClick={(): void => filterByGroup()}
+                >
+                  <FilterListIcon fontSize="small" />
+                </IconButton>
+                <IconButton
+                  aria-label={`${group.name} Configure`}
+                  size="small"
+                  style={{ color: theme.palette.common.white }}
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx:101]: Same comment as above

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+                  title={t(labelConfigure)}
+                  onClick={(): void => {
+                    window.location.href = group.configuration_uri as string;
+                  }}
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx:105]: Extract it to a specific function

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/Groups.tsx [hunk 15]
```diff
@@ -1,46 +1,47 @@
...
 
 interface Props {
-  details: ResourceDetails;
+  details: ResourceDetails | undefined;
 }
 
+const useStyles = makeStyles((theme) => ({
+  groups: {
+    display: 'flex',
+    padding: theme.spacing(1, 1, 1, 1),
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/Groups.tsx:23]:
```suggestion
    padding: theme.spacing(1),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/index.tsx [hunk 16]
```diff
@@ -10,7 +10,7 @@ const useStyles = makeStyles((theme) => ({
     color: theme.palette.success.main,
   },
   container: {
-    height: 65,
+    height: 70,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/index.tsx:13]: Do we still need this?

**Review by callapa — DISMISSED** [review]

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/index.tsx (hunk 16 above)
Comment (Jordan91) [reply www/front_src/src/Resources/Details/tabs/Details/DetailsCard/index.tsx:13]: No , we can come back with 65 for height , by using isCustomCard , issue is solved

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx (hunk 14 above)
Comment (Jordan91) [reply www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx:105]: Ok , i'll add this function
const configureGroup = (): void => {
                    window.location.href = group.configuration_uri as string;
                  };

Comment (Thebarda) [reply www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx:105]: Maybe goToConfigurationPage

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+  },
+  chipLabelColor: { color: true ? 'transparent' : 'unset' },
+}));
+
+interface GroupsChipProps {
+  group: Group;
+  type: string;
+}
+
+const GroupChip = ({ group, type }: GroupsChipProps): JSX.Element => {
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx:50]:
```suggestion
interface Props {
  group: Group;
  type: string;
}

const GroupChip = ({ group, type }: Props): JSX.Element => {
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx [hunk 18]
```diff
@@ -0,0 +1,128 @@
...
+  chipIconColor: {
+    color: theme.palette.common.white,
+  },
+  chipLabel: {
+    display: 'grid',
+    justifyItems: 'center',
+    minWidth: theme.spacing(7),
+    overflow: 'hidden',
+  },
+  chipLabelColor: { color: 'transparent' || 'unset' },
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx:43]:
```suggestion
  chipLabelColor: { color: 'transparent' },
```
`unset` is not used

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx [hunk 19]
```diff
@@ -0,0 +1,128 @@
...
+              </Typography>
+            </Tooltip>
+            {isHovered && (
+              <Grid className={classes.chipHover}>
+                <IconButton
+                  aria-label={`${group.name} Filter`}
+                  className={classes.chipIconColor}
+                  size="small"
+                  title={t(labelFilter)}
+                  onClick={(): void => filterByGroup()}
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/GroupChip.tsx:104]:
```suggestion
                  onClick={filterByGroup}
```

**Review by Thebarda — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**